PyMca5/PyMcaCore/TiledDataSource.py

```diff
@@ -0,0 +1,30 @@
+import copy
+
+from .DataObject import DataObject
+
+
+SOURCE_TYPE = "Tiled"
+
+
+class TiledDataSource(object):
+    """Manages a Tiled CatalogOfBlueskyRuns as a PyMca Data Source"""
+    def getDataObject(self, key: str, selection=None):
+        """Return data associated with 'key' and optional 'selection'."""
+        data = DataObject()
+        labels = ()
+        if selection is not None:
+            labels = selection["Channel List"].copy()
+        data.info = {
+            # For now, only allow one-dimensional data arrays
+            "selectiontype": "1D",
+            "selection": copy.deepcopy(selection),
```

Comment:
I gather we're going to mutate this and we're deep-copying to avoid mutating user inputs? Some comment on the deepcopy over the type of `selection` may be useful here.

Reply:
That's a good idea.  FWIW, I could just leave this field undefined, and the next step in PyMCA plotting will create it with this same operation.  I haven't dug too deep yet, but it looks like multiple (independent) copies of "selection" get used by different parts of the app (that have access to different parts of the the parent "record" that gets passed via signals).

Reply:
Comment added in commit `20461057`